### PR TITLE
udev: fix multi match

### LINKED ISSUE #13518: missing evdev symlinks since v243
**systemd version the issue has been seen with**
> 243

**Used distribution**
 > Arch Linux x86_64 / Kernel: 5.2.13-arch1-1-ARCH

**Expected behaviour you didn't see**
 ```
ls -alh /dev/input/by-id 
total 0
drwxr-xr-x 2 root root 240 10 sept. 11:05 .
drwxr-xr-x 4 root root 680 10 sept. 11:05 ..
lrwxrwxrwx 1 root root  10 10 sept. 10:51 usb-Corsair_Components__Inc._Corsair_Gaming_MM800_RGB_POLARIS_1E1001060115BDAE61372656441900F5-event-if01 -> ../event11
lrwxrwxrwx 1 root root   9 10 sept. 10:51 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-event-if01 -> ../event4
lrwxrwxrwx 1 root root   9 10 sept. 10:51 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-event-kbd -> ../event3
lrwxrwxrwx 1 root root   9 10 sept. 10:51 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-if01-event-kbd -> ../event6
lrwxrwxrwx 1 root root   9 10 sept. 10:51 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-if01-event-mouse -> ../event7
lrwxrwxrwx 1 root root   9 10 sept. 10:51 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-if01-mouse -> ../mouse0
lrwxrwxrwx 1 root root   9 10 sept. 10:51 usb-FiiO_DigiHug_USB_Audio-event-if00 -> ../event8
lrwxrwxrwx 1 root root  10 10 sept. 10:51 usb-Logitech_Gaming_Mouse_G502_078D36613633-event-mouse -> ../event13
lrwxrwxrwx 1 root root  10 10 sept. 10:51 usb-Logitech_Gaming_Mouse_G502_078D36613633-if01-event-kbd -> ../event14
lrwxrwxrwx 1 root root   9 10 sept. 10:51 usb-Logitech_Gaming_Mouse_G502_078D36613633-mouse -> ../mouse1
```

**Unexpected behaviour you saw**

```
ls -alh /dev/input/by-id
total 0
drwxr-xr-x 2 root root 200 10 sept. 10:15 .
drwxr-xr-x 4 root root 700 10 sept. 10:15 ..
lrwxrwxrwx 1 root root  10 10 sept. 09:54 usb-Blue_Microphones_Yeti_Stereo_Microphone_REV8-event-if03 -> ../event11
lrwxrwxrwx 1 root root  10 10 sept. 09:54 usb-Corsair_Components__Inc._Corsair_Gaming_MM800_RGB_POLARIS_1E1001060115BDAE61372656441900F5-event-if01 -> ../event12
lrwxrwxrwx 1 root root   9 10 sept. 09:54 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-event-if01 -> ../event5
lrwxrwxrwx 1 root root   9 10 sept. 09:54 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-if01-event-kbd -> ../event6
lrwxrwxrwx 1 root root   9 10 sept. 09:54 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-if01-event-mouse -> ../event7
lrwxrwxrwx 1 root root   9 10 sept. 09:54 usb-Das_Keyboard_5Q_RGB_Mechanical_Keyboard_0000000000000000-if01-mouse -> ../mouse0
lrwxrwxrwx 1 root root   9 10 sept. 09:54 usb-FiiO_DigiHug_USB_Audio-event-if00 -> ../event8
lrwxrwxrwx 1 root root  10 10 sept. 09:54 usb-Logitech_Gaming_Mouse_G502_078D36613633-if01-event-kbd -> ../event27
```

**Steps to reproduce the problem**

upgrade from v242 to v243

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #13518.

### CODE REVIEW

**Comment by yuwata** [comment]
@keszybz Please backport this to v243-stable after it is accepted.

**Comment by evverx** [comment]
@yuwata `tools/oss-fuzz.sh` is failing on Pipelines and Travis CI. I'll restart the failed jobs once I make sure https://github.com/systemd/systemd/pull/13523 is enough to fix this and merge that PR.

**Comment by yuwata** [comment]
@evverx Thanks a lot!

File: src/udev/udev-rules.c [hunk 1]
```diff
@@ -431,35 +433,30 @@ static int rule_line_add_token(UdevRuleLine *rule_line, UdevRuleTokenType type,
...
+                        }
+                        *b = '\0';
+                        if (bar)
+                                empty = true;
+
+                        if (empty) {
+                                if (match_type == MATCH_TYPE_GLOB)
+                                        match_type = MATCH_TYPE_GLOB_WITH_EMPTY;
+                                if (match_type == MATCH_TYPE_PLAIN)
+                                        match_type = MATCH_TYPE_PLAIN_WITH_EMPTY;
```
Comment (keszybz) [inline src/udev/udev-rules.c:459]: This should work. Alternatively, we could simply switch to strvs.

**Pull request merged**